### PR TITLE
connect ground-height and resolution arguments to runtime config

### PR DESCRIPTION
**Pull request by j9ac9k**

### CODE REVIEW

**Comment by hobu** [comment]
https://github.com/NCALM-UH/CODEM/blob/dc170ac913623bd14b015a743d5d3ea285624f47/src/vcd/preprocessing/preprocess.py#L180 and https://github.com/NCALM-UH/CODEM/blob/dc170ac913623bd14b015a743d5d3ea285624f47/src/vcd/preprocessing/preprocess.py#L196 needs to be threaded back to an option. This is the clustering tolerance.

**Comment by hobu** [comment]
https://github.com/NCALM-UH/CODEM/blob/dc170ac913623bd14b015a743d5d3ea285624f47/src/vcd/meshing/mesh.py#L62 should probably be added as a new option. This is a list of ClusterID values that should not be included when producing mesh blobs.

**Comment by hobu** [comment]
We should probably add a ground/notground attribute to the shapefile output too https://github.com/NCALM-UH/CODEM/blob/dc170ac913623bd14b015a743d5d3ea285624f47/src/vcd/meshing/mesh.py#L81

**Comment by j9ac9k** [comment]
> https://github.com/NCALM-UH/CODEM/blob/dc170ac913623bd14b015a743d5d3ea285624f47/src/vcd/preprocessing/preprocess.py#L180
> 
> and
> https://github.com/NCALM-UH/CODEM/blob/dc170ac913623bd14b015a743d5d3ea285624f47/src/vcd/preprocessing/preprocess.py#L196
> 
> needs to be threaded back to an option. This is the clustering tolerance.

Implemented in 2e47f98

> https://github.com/NCALM-UH/CODEM/blob/dc170ac913623bd14b015a743d5d3ea285624f47/src/vcd/meshing/mesh.py#L62
> 
> should probably be added as a new option. This is a list of ClusterID values that should not be included when producing mesh blobs.

Implmeneted in 8e0d589

**Comment by j9ac9k** [comment]
> We should probably add a ground/notground attribute to the shapefile output too
> 
> https://github.com/NCALM-UH/CODEM/blob/dc170ac913623bd14b015a743d5d3ea285624f47/src/vcd/meshing/mesh.py#L81

Looks like to add the attribute it would be 

```python
w.field("ground", "L")
```

How do I determine if it is ground or not ground though?

File: src/vcd/meshing/mesh.py [hunk 1]
```diff
@@ -90,4 +91,5 @@ def write(self, filename: str, clusters: List[trimesh.Trimesh]) -> None:
                     cluster.triangles,
                     partTypes=[TRIANGLE_STRIP] * len(cluster.triangles),
                 )  # one type for each part
-                w.record(cluster.volume, cluster.area, cluster.cluster_id)
+                is_ground = cluster.cluster_id == 2
```
Comment (hobu) [inline src/vcd/meshing/mesh.py:94]: I think is_ground is true whether or not the points have Classification==2, not their cluster_id is 2. That's just the 2nd cluster that was made.

**Review by hobu — CHANGES_REQUESTED** [review]
Fix the is_ground thing and it looks good to me.

File: src/vcd/meshing/mesh.py (hunk 1 above)
Comment (j9ac9k) [reply src/vcd/meshing/mesh.py:94]: 🤦🏻‍♂️ why yes.... the classification, not the cluster_id .... ugh yeah I'll fix that.

Comment (j9ac9k) [reply src/vcd/meshing/mesh.py]: Fixed in d03b0903901a923bd2fa28fdb059e7b77a352133

**Pull request merged**